engine/crates/engine-v2/src/execution/header_rule.rs

```diff
@@ -0,0 +1,123 @@
+use std::{borrow::Cow, str::FromStr};
+
+use http::header;
+use schema::{HeaderRuleWalker, NameOrPatternRef};
+
+use crate::engine::RequestContext;
+
+static FORWARD_BLACKLIST: [header::HeaderName; 12] = [
+    header::ACCEPT,
+    header::ACCEPT_CHARSET,
+    header::ACCEPT_ENCODING,
+    header::ACCEPT_CHARSET,
+    header::ACCEPT_LANGUAGE,
+    header::ACCEPT_RANGES,
+    header::CONNECTION,
+    header::HeaderName::from_static("keep-alive"),
+    header::CONTENT_LENGTH,
+    header::CONTENT_TYPE,
+    header::HOST,
+    header::USER_AGENT,
+];
```

Comment:
There's a few other hop-by-hop headers in the spec that we should probably omit: https://datatracker.ietf.org/doc/html/rfc2616#section-13.5.1

Might also dispute user-agent.  Can see reasons to not include it, but could also see reasons to include it?  I'm not sure...

Reply:
they can always forward it with a different name, so sounded fair to me 🤷 
Not critical, but felt more "sincere" for a service to use a proper user agent

Reply:
good catch on the hop-by-hop headers!